frontend/webadmin/modules/gwt-common/src/main/java/org/ovirt/engine/ui/common/widget/uicommon/storage/DisksAllocationView.java

```diff
@@ -189,11 +190,15 @@ private void initListeners(final DisksAllocationModel model) {
 
     void addDiskList(DisksAllocationModel model) {
         diskListPanel.clear();
-        diskAllocationLabel.setVisible(!model.getDisks().isEmpty());
+        List<DiskModel> disks = model.getDisks();
+        if (disks == null) {
```

Comment:
Maybe the better way was to fix the `DisksAllocationModel`, so the disks will be set to `new ArrayList<>()` there (in `disks` data member initialization and `setDisks()`, or, IMHO less recommended, when returning `null` by `getDisks()`)?
Otherwise who knows where NPE can be received by another `model.getDisks()` users or even directly in `DisksAllocationModel` methods, it.e., currently in (almost) all places `DisksAllocationModel` the `null` check is performed, but in 2 methods it's missed: `updateDisks()` & `getDisksToOperate()`, so potentially there are more NPEs.

Reply:
yes, I agree that it could have been better to replace 'null' with empty list in the model but iI don't think it's cost-effective to change it now - the rest of the code already knows to check for `null` where needed, it's just this new code that was recently introduced that didn't take it into account..

Reply:
OK,
1) As I wrote IMHO there is still an issue with `DisksAllocationModel::getDisksToOperate()` that can result in NPE.
2) Suggest to change `new ArrayList<>()` to either `Collections.emptyList()` or `Collections.EMPTY_LIST`. More elegant and serialization/immutable issues do not exist in this case :)

Reply:
> 1. As I wrote IMHO there is still an issue with `DisksAllocationModel::getDisksToOperate()` that can result in NPE.

I believe that if we'll track the calls to `DisksAllocationModel::getDisksToOperate()`, we'll find that the disks are always set when get there (or it's called from an event listener that cannot be executed for diskless VMs), otherwise we would have seen such NPEs before..

> 2. Suggest to change `new ArrayList<>()` to either `Collections.emptyList()` or `Collections.EMPTY_LIST`. More elegant and serialization/immutable issues do not exist in this case :)

ack, yeah, this variable is not sent to the backend, changed